### PR TITLE
feat: change api call to get dashboard data and listing of data

### PR DESCRIPTION
**Pull request by ktx-vaidehi**
- #5196

## Summary by CodeRabbit

- **New Features**
	- Enhanced dashboard data fetching and error handling for improved user experience.
	- Introduced a new reactive variable to manage the fetched dashboards more effectively.
	- Added a new method for retrieving dashboard data by ID.
	- Improved form validation for drilldown operations.
  
- **Bug Fixes**
	- Enhanced error handling logic to display appropriate messages based on dashboard data retrieval conditions.
  
- **Chores**
	- Refactored various functions for better maintainability and clarity in dashboard management.

### CODE REVIEW

File: web/src/components/dashboards/PanelSchemaRenderer.vue [hunk 1]
```diff
@@ -810,9 +810,13 @@ export default defineComponent({
...
-          const dashboardData = allDashboardData.find(
-            (dashboard: any) => dashboard.title == drilldownData.data.dashboard,
-          );
+
+          const dashboardId = allDashboardData?.find(
+            (dashboard: any) =>
+              dashboard.title === drilldownData.data.dashboard
+          )?.dashboardId;
+
+          const dashboardData = await getDashboard(store, dashboardId, folderId);
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/components/dashboards/PanelSchemaRenderer.vue:819]: this error handling is not required

File: web/src/components/dashboards/addPanel/DrilldownPopUp.vue [hunk 2]
```diff
@@ -486,20 +487,29 @@ export default defineComponent({
...
+        tabList.value = [];  
+        return;  
+      }  
 
+      // get dashboard data
+      // by using dashboard name, find dashboard data
+      const dashboardData = await getDashboard(store, dashboardId, folderData?.folderId);
+      
       // if no dashboard with same dashboard name found, return
       if (!dashboardData) {
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/components/dashboards/addPanel/DrilldownPopUp.vue:514]: fixed

File: web/src/views/Dashboards/Dashboards.vue [hunk 3]
```diff
@@ -537,16 +537,30 @@ export default defineComponent({
...
+          dismiss();
+          searchAcrossFolders.value = false;
+          router.push({
+            path: "/dashboards",
+            query: {
+              org_identifier: store.state.selectedOrganization.identifier,
+              folder: activeFolderId.value,
+            },
+          });
+        }
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/views/Dashboards/Dashboards.vue:563]: it is already present

File: web/src/stores/index.ts [hunk 4]
```diff
@@ -126,6 +127,9 @@ export default createStore({
     setAllDashboardList(state, payload) {
       state.organizationData.allDashboardList = payload;
     },
+    setDashboardData(state, payload) {
+      state.organizationData.allDashboardData = payload;
+    },
```
[the remark this answers is not in the document]
Comment (ktx-kirtan) [reply web/src/stores/index.ts:132]: We replace all the data when we are sending it. So, this is not required.

File: web/src/stores/index.ts [hunk 5]
```diff
@@ -236,6 +240,9 @@
     setAllDashboardList(context, payload) {
       context.commit("setAllDashboardList", payload);
     },
+    setDashboardData(context, payload) {
+      context.commit("setDashboardData", payload);
+    },
```
[the remark this answers is not in the document]
Comment (ktx-kirtan) [reply web/src/stores/index.ts:245]: We will be adding the types in another PR

**Review by ktx-kirtan — APPROVED** [review]

**Pull request merged**